WebDevStudios/nextjs-wordpress-starter
# Feature/menus

Closes #4 

### DESCRIPTION ###
Added support for native WP Menus (Primary and Mobile) of the `wd_s` theme.

A `<Navigation/>` component was created to help render the menus. The component has added to both the `<Header/>` and `<Footer/>`.

**Notes**: 

- The Mobile Menu is not currently being rendered as the proper components are yet to be built.
- When pulling in a WP Menu, the `url` node of the query contains the full path to the WP install. I handled this by adding a helper function (`convertWPURL)` that replaces the `CMS_URL` constant.
- The GraphQL query is set up to pull two levels of navigation, however only the top level is being display. Added a drop menu will require additional functionality(CSS/JS) to pass a11y requirements and should likely be moved to a new issue imo.

### SCREENSHOTS ###

![ZpZR_CYQ](https://user-images.githubusercontent.com/428624/90928040-2578f280-e3c4-11ea-8865-ee4b834c6fa4.png)

### OTHER ###
- [x] Is this issue accessible? (Section 508/WCAG 2.1AA)
- [x] Does this issue pass all the linting?

### STEPS TO VERIFY ###

1. Update the `Primary` Menu on https://nextjs.wpengine.com
2. Rebuild the NextJS app
3. View the build to verify the menu update (header & footer)

### DOCUMENTATION ###
No documentation updates are required for this feature.
